src/ostree/ot-builtin-static-delta.c

```diff
@@ -438,13 +446,63 @@ ot_static_delta_builtin_apply_offline (int argc, char **argv, OstreeCommandInvoc
       return FALSE;
     }
 
+#if defined(HAVE_LIBSODIUM)
+  /* Initialize crypto system */
+  opt_sign_name = opt_sign_name ?: OSTREE_SIGN_NAME_ED25519;
```

Comment:
Hmm...we don't have this default elsewhere I don't think.  Is it really necessary?  In other code the default comes from the repository configuration or the CLI and that's it.

Reply:
This is initialized this way in [ostree_builtin_commit()](https://github.com/ostreedev/ostree/blob/master/src/ostree/ot-builtin-commit.c#L879), [ostree_builtin_sign()](https://github.com/ostreedev/ostree/blob/master/src/ostree/ot-builtin-sign.c#L108) and [ostree_builtin_summary()](https://github.com/ostreedev/ostree/blob/master/src/ostree/ot-builtin-summary.c#L106).

Iiuc, the default crypto system is only defined in configuration file for remote repositories.
When we try to apply a static delta offline, there's no reference to a remote repository.